packages/app/src/app/hex-color/hex-color.service.ts

```diff
@@ -14,20 +14,19 @@ export class HexColorService {
       {
         hooks: [
           {
-            name: 'validate-hex',
             after: (
               hookContext,
-              evaluationDetails: FlagEvaluationDetails<string>
+              evaluationDetails: EvaluationDetails<string>
             ) => {
               // validate the hex value.
               const hexPattern = /[0-9A-Fa-f]{6}/g;
-              if (hexPattern.test(evaluationDetails.value)) {
-                return evaluationDetails.value;
-              } else {
+              if (!hexPattern.test(evaluationDetails.value)) {
                 console.warn(
                   `Got invalid flag value '${evaluationDetails.value}' for ${hookContext.flagKey}, returning ${hookContext.defaultValue}`
                 );
-                return hookContext.defaultValue;
+                throw new Error(
```

Comment:
Maybe we should add a comment here explaining this will cause a default. It's a bit less obvious than the previous return, even though I think it's a better pattern.